worker/src/service.rs

```diff
@@ -0,0 +1,58 @@
+use js_sys::Promise;
+use wasm_bindgen::{JsCast, JsValue};
+use wasm_bindgen_futures::JsFuture;
+
+use crate::{env::EnvBinding, Request, Response, Result};
+
+pub struct Service {
```

Comment:
Same with this, should be able to merge this with the existing [`Fetcher`](https://github.com/cloudflare/workers-rs/blob/main/worker/src/fetcher.rs#L8).